Jenkinsfile

```diff
@@ -110,6 +113,9 @@ def job = {
     } else {
         // For non-release builds, we append the build number to the maven artifacts and docker image tag
         config.ksql_db_artifact_version = config.ksql_db_version + '-rc' + env.BUILD_NUMBER
+        // For non-release builds, we upload the RC artifacts to a different prefix for testing.
+        // Actual release artifacts will go to a "prod" prefix.
+        config.ksql_db_packages_prefix = config.ksql_db_packages_prefix + '-rcs'
```

Comment:
Its a "sin" you guys are copying RC jars to ksqldb-maven, I don't think theres an easy solutuion there for maven, but this is to prevent us from repeating that sin for OS packages. Here is a breakdown:

(say the release is 0.15.0)
* Non-release builds will deploy to `{deb,rpm,archive}/0.15-rcs`
* Actual release builds will deploy to `{deb,rpm,archive}/0.15`
* Promote will only ever move `{deb,rpm,archive}/0.15` to a production bucket, omitting any "rcs" that were generated in that time.